src/Model/Sneak.re

```diff
@@ -0,0 +1,120 @@
+open Revery.Math;
+
+type callback = unit => unit;
+type bounds = unit => option(Rectangle.t);
+
+type sneakInfo = {
+  callback,
+  boundingBox: BoundingBox2d.t,
+};
+
+type sneak = {
+  callback,
+  boundingBox: BoundingBox2d.t,
+  id: string,
+};
+
+[@deriving show({with_path: false})]
+type action =
+  | Initiated
+  | Discover([@opaque] list(sneakInfo))
+  | KeyboardInput(string)
+  | Stopped;
```

Comment:
You're still treating these basically as functions, despite naming them as events. The events described haven't occurred yet, but are assumed to be the result of processing these, hence essentially begging the question. This is going to cause confusion if you try to debug based on this information, assuming something has happened becasue the log says ti has when it fact it hasn't.

Let functions be functions. `Initiated` and `Stopped` can be ordinary functions, `start` and `stop`, taking and returning a `Sneak.t`, and be invoked from the actual events that should trigger it, `SomeButtonClicked` or `Command("sneak.stop")`.

`Discover` could just be named `Discovered` I think.

Reply:
Thanks! Still need to work on this.

I've updated the PR and refactored the actions a bit - hopefully it's a step in the right direction.

Reply:
Definitely, but `NoneAvailable` certainly isn't a very event-like name. Couldn't simply `Completed` work?